src/main/common/huffman.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "huffman.h"
+
+
+int huffmanEncodeBuf(uint8_t *outBuf, int outBufLen, const uint8_t *inBuf, int inLen, const huffmanTable_t *huffmanTable)
+{
+    int ret = 0;
+
+    uint8_t *outByte = outBuf;
+    *outByte = 0;
+    uint8_t outBit = 0x80;
+    
+    for (int ii = 0; ii < inLen; ++ii) {
+        const int huffCodeLen = huffmanTable[*inBuf].codeLen;
+        const uint16_t huffCode = huffmanTable[*inBuf].code;
+        ++inBuf;
```

Comment:
Here is quite naive implementation that is packing bits in one go. Intermediate store of result back into buffer can be optimized by caching `v` and `used` ... 
```
int pack(void* dst, int dbits, int val, int bits)
{
    typedef uint32_t it;
#define IB (sizeof(it) * 8)
    while(bits) {
        const int used = dbits % IB;
        it v;
        if(used)
            v = ((it*)dst)[dbits / IB];
        else
            v = 0;
        const int chunk = MIN((int)IB - used, bits);
        const it add = val & ((1 << chunk) - 1);
        val >>= chunk;
        v |= add << used;
        ((it*)dst)[dbits / IB] = v;
        dbits += chunk; bits -= chunk;
    }
    return dbits;
}